sdcm/sct_config.py

```diff
@@ -2054,51 +2400,67 @@ def get_config_option(cls, name):
     def get_default_value(self, key, include_backend=False):
 
         default_config_files = [sct_abs_path('defaults/test_default.yaml')]
-        backend = self['cluster_backend']
-        if backend and include_backend:
-            default_config_files += self.defaults_config_files[str(backend)]
+        if self.cluster_backend and include_backend:
+            default_config_files += self.defaults_config_files[str(self.cluster_backend)]
 
         return anyconfig.load(list(default_config_files)).get(key, None)
 
     def _load_environment_variables(self):
         environment_vars = {}
-        for opt in self.config_options:
-            if opt['env'] in os.environ:
+        for field_name, field in self.model_fields.items():
+            if field.exclude or not field.json_schema_extra:
+                continue
+
+            field_env = field.json_schema_extra.get('env')
+
+            def no_op(x):
+                return x
+
+            for annotation in field.metadata:
+                if isinstance(annotation, BeforeValidator):
+                    from_env_func = annotation.func
+                    break
+            else:
+                from_env_func = no_op
```

Comment:
are you sure no_op is right here? Env vars will always return strings - shouldn't be casted to annotated one?

Reply:
this is when there is no annotation we can work with

Reply:
so every annotation we'll have can be assumed to convert string to anything.

for example if I put there other pydantic structure ? or `BaseNode` (just example)